packages/pipeline-editor/src/PipelineEditorWidget.tsx

```diff
@@ -949,15 +951,27 @@ export class PipelineEditor extends React.Component<
    * if there are invalid properties. If there are none,
    * returns null.
    */
-  validateProperties(node: any): string {
+  async validateProperties(node: any): Promise<string> {
+    const notebookValidationErr = await this.app.serviceManager.contents
+      .get(node.app_data.filename)
+      .then((result: any): any => {
+        return null;
+      })
+      .catch((err: any): any => {
+        return 'notebook does not exist';
+      });
+    let validationError = notebookValidationErr;
     if (
       node.app_data.runtime_image == null ||
       node.app_data.runtime_image == ''
     ) {
-      return 'no runtime image.';
-    } else {
-      return null;
+      if (validationError != '') {
+        validationError += ' and no runtime image.';
```

Comment:
though this works now, this concatenation will not scale as we add validation. I would suggest instead keeping each error in an array of strings and displaying each on a new line when shown

Reply:
To clarify a bit on this: This function can still return a single string, I just think that each validation check should add to an array of errors then the errors array can be delisted with new line prior to return